### ISSUE TITLE
Error not logged by "throw NotFoundException" 

### ISSUE THREAD

**Issue by cmoulliard** [halkyonio/primaza-poc#379]
## Issue

The following [line](https://github.com/halkyonio/primaza-poc/blob/3347ce91fc806e80677c39e906cd507ac2593489/app/src/main/java/io/halkyon/resource/page/ApplicationResource.java#L122) do not throw an exception if the "claim.application" is null. Why ?

```
    @Transactional
    @POST
    @Produces(MediaType.TEXT_HTML)
    @Path("/claim/{id}")
    public Response doClaimApplication(@PathParam("id") long applicationId, @FormParam("claimId") long claimId) {
        LOG.debug("Application claiming started ...");
        Application application = Application.findById(applicationId);
        if (application == null) {
            throw new NotFoundException(String.format("Application %s not found", applicationId));
        }
        LOG.debugf("Application found: %s", application.name);

        Claim claim = Claim.findById(claimId);
        if (claim == null) {
            throw new NotFoundException(String.format("Claim %s not found", claimId));
        }
        if (claim.service == null) {
            throw new NotAcceptableException(String.format("Claim %s has no services available", claimId));
        }
        LOG.debug("Claim is not null like claim.service");
        if (claim.service.installable) {
            LOG.debugf("Service is installable: %s", claim.service.installable);
            if (claim.application == null) {
                claim.application = application;
                // TODO: If we try a NotFoundException, then it is not displayed within the console of the application !
                // throw new NotFoundException("Claim with name " + claim.name + " is not associated to an
                // application.");
            }
```

Moreover, when this curl request is issued, it receives as reponse a HTTP 404 error and not a HTTP 500
```
./scripts/data/bind_application.sh application_name=atomic-fruits claim_name=fruits-claim
NOTE: Searching about the application to be bound ...
NOTE: curl -H 'Accept: application/json' -s primaza.127.0.0.1.nip.io/applications/name/atomic-fruits
NOTE: Application ID to be bound: 1
NOTE: Searching about the claim ...
NOTE: Claim ID to be bound: 1
NOTE: curl -X POST -H 'Content-Type: application/x-www-form-urlencoded' -d "claimId=1" -s -i primaza.127.0.0.1.nip.io/applications/claim/1
ERROR: Application failed to be bound in Primaza: 404 Not found
```

**Comment by Sgitario** [comment]
> The following [line](https://github.com/halkyonio/primaza-poc/blob/3347ce91fc806e80677c39e906cd507ac2593489/app/src/main/java/io/halkyon/resource/page/ApplicationResource.java#L122) do not throw an exception if the "claim.application" is null. Why ?

Before, it was throwing a NotFoundException, and you commented it out. 
From my point of view, if we have the application and the claim, we can bind it, so I see no reasons to throw an exception and the current code is correct (but the comments should be removed).

**Comment by Sgitario** [comment]
> Moreover, when this curl request is issued, it receives as reponse a HTTP 404 error and not a HTTP 500

Why should it return an HTTP 500 ?

**Comment by cmoulliard** [comment]
> Before, it was throwing a NotFoundException, and you commented it out.

I commented the 2 two lines to show the issu. So if `claim.application == null`, then an error is throwed BUT it  don't appear within the log of the quarkus application. The client issuing the request (curl, ...) gets an error HTTP 404 but nobody knows why an error happened here. This is why I opened the ticket ;-)

**Comment by cmoulliard** [comment]
> Why should it return an HTTP 500 ?

As an error has ben thrown `throw new NotFoundException("Claim with name " + claim.name + " is not associated to an application.");`

**Comment by Sgitario** [comment]
I don't know why `if claim.application == null, then an error is thrown` was done, but it seems reasonable to me that instead of returning an error (it should be neither a 404 as was done before, nor a 500), simply sets the `claim.application = application`.

**Comment by cmoulliard** [comment]
This is what I did. The purpose of the ticket is still relevant. Why, when an error is throw, such as `throw new NotFoundException()`, it does not appear within the log of the quarkus application and subsequently, why no HTTP 500 is then generated ?

**Comment by Sgitario** [comment]
> This is what I did. The purpose of the ticket is still relevant. Why, when an error is throw, such as `throw new NotFoundException()`, it does not appear within the log of the quarkus application and subsequently, why no HTTP 500 is then generated ?

Why subsequently should it throw the HTTP 500 exception? The `NotFoundException` exception is mapped to the 404 HTTP code.

**Comment by cmoulliard** [comment]
> The `NotFoundException` exception is mapped to the 404 HTTP code.

Ok but as nothing appear within the quarkus log, then it is impossible to know why we got a HTTP 404 error.

**Comment by Sgitario** [comment]
> Why, when an error is throw, such as `throw new NotFoundException()`, it does not appear within the log of the quarkus application

@geoand just clarified this, you need to add this property:

```
quarkus.log.category."org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext".level=DEBUG
```

More info in: https://quarkus.io/guides/resteasy-reactive#exception-mapping

**Comment by aureamunoz** [comment]
Fixed by efc000f33308c66a76fa34d5d90f78a2e6b5b53b
